src/Channels.Networking.Sockets/SocketConnection.cs

```diff
@@ -0,0 +1,304 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Channels.Networking.Sockets
+{
+    public class SocketConnection : IChannel
+    {
+        private Socket _socket;
+        private Channel _input, _output;
+        private ChannelFactory _channelFactory;
+        private readonly bool _ownsChannelFactory;
+
+        static readonly EventHandler<SocketAsyncEventArgs> _asyncCompleted = OnAsyncCompleted;
+
+        public IReadableChannel Input => _input ?? CreateInput();
```

Comment:
This was quite deliberate, but I don't know if there's a single "right" answer; my intent here was not to alloc/buffer from external input until someone expressed an interest in input. Might be Imagineering, though.